src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -372,18 +368,24 @@ class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
     };
 
     handleSettingsSubmit = () => {
-        const {onChange, value = []} = this.props;
+        const {onChange} = this.props;
+        const oldValues = this.value || [];
 
         const {blockSettingsFormStore, blockSettingsOpen} = this;
 
-        if (!blockSettingsFormStore || blockSettingsOpen === undefined || !value) {
+        if (!blockSettingsFormStore || blockSettingsOpen === undefined || !oldValues) {
             return;
         }
 
+        const newValues = toJS(oldValues);
+        jsonpointer.set(newValues[blockSettingsOpen], '/' + SETTINGS_KEY, blockSettingsFormStore.data);
+
+        this.setValue(newValues);
+
         onChange([
-            ...value.slice(0, blockSettingsOpen),
-            {...value[blockSettingsOpen], [SETTINGS_KEY]: blockSettingsFormStore.data},
-            ...value.slice(blockSettingsOpen + 1),
+            ...newValues.slice(0, blockSettingsOpen),
+            {...newValues[blockSettingsOpen]},
+            ...newValues.slice(blockSettingsOpen + 1),
         ]);
```

Comment:
Now the `toJS` and the `onChange` call create a new array, which kind of is doubling the work :see_no_evil: I think I would have either just called `onChange(newValues)`, or would have rewritten that differently (that would be my favourite approach):

```javascript
const newValues = [
    ...oldValues.slice(0, blockSettingsOpen),
    {...oldValues[blockSettingsOpen], [SETTINGS_KEY]: blockSettingsFormStore.data},
    ...oldValues.slice(blockSettingsOpen +1),
];

this.setValue(newValues);
onChange(newValues);
```

This would make the call to `toJS` and `jsonpointer.set` obsolete.